api/models/api_models/calendar_event.js

```diff
@@ -0,0 +1,89 @@
+import { BaseModel, BaseModelUpdate, string, date, now, boolean } from "./base_model.js";
+import Accounts from "../models/enums/accounts.js";
+import { UserPublicResponse } from "./user.js";
+
+/**
+ * Represents the http request body required
+ * to create a calendar event in the database.
+ */
+export class CalendarEventCreate extends BaseModel {
+  static schema = {
+    title: { type: string, required: true },
+    description: { type: string },
+    startTime: { type: date, required: true },
+    endTime: { type: date, required: true },
+    location: { type: string },
+    public: { type: boolean, required: true, default: false },
+    visibleToRoles: { type: [string], enum: Accounts.listStr, required: true },
```

Comment:
To get enumerated types as a list, use the enum `listr()` method. For example here, to list account type as a list of strings, do:
```js
Accounts.listr()
```